src/distribution.cpp

```diff
@@ -105,9 +105,9 @@ size_t DiscreteIndex::sample(uint64_t* seed) const
 void DiscreteIndex::normalize()
 {
   // Renormalize density function so that it sums to unity
-  double norm = std::accumulate(prob_.begin(), prob_.end(), 0.0);
+  integral_ = std::accumulate(prob_.begin(), prob_.end(), 0.0);
```

Comment:
Do we need to set `integral_ = 1.0` after looping through all the probability values and normalizing them? If I'm following correctly, an un-normalized probability distribution would end up with a value of `integral_` different from 1. If you then called `normalize()` again the value of `_integral` would change while the underlying probability density values would not, which I would think is undesirable.

Reply:
The point of `integral_` is to preserve the integral of the original distribution as it was read in. The actual probabilities in `prob_` get normalized for sampling purposes, but we save the original sum so that if the discrete distribution is used as part of a mixture, we still know what the original integral (i.e., the intensity of the discrete source) was.

You are correct that if you were to call `normalize()` again, it would change the value of the integral. This is why the `normalize()` method is not public.

Reply:
Ah, ok I missed that it wasn't public. I think I'd like to see a little extra documentation on `integral_` when it takes on this meaning, because the other distributions just return 1 and `integral()` is public.

Reply:
Ok, I added an extra sentence above here explaining its purpose a little more. If you had something else in mind, let me know!

Reply:
okee doke, at least my small brain won't freak out next time I read this part of the code.